qark/modules/findPending.py

```diff
@@ -21,456 +22,499 @@
-								found=True
-								for a in c.initializer.arguments:
-									if type(a) is m.ClassLiteral:
-										explicit=True
-										results=[found,explicit]
-										return results
-								explicit=False
-								results=[found,explicit]
-								return results
-			else:
-				if hasattr(current,'_fields'):
-					for r in current._fields:
-						results=intent_digger(getattr(current,r),intent)
-						if results[0] == True:
-							return results
-							
-	elif type(t) is list:
-		for y in t:
-			results=intent_digger(y,intent)
-			if results[0] == True:
-				return results
-
-	return results
+def start(queue, height):
+    """ Start finding pending intents """
+    results = []
+    global tree
+    global current_file
+    count = 0
+
+    # TODO - Look for use of fillIn method which can make this a much more exploitable condition
+    for j in common.java_files:
+        count = count + 1
+        pub.sendMessage('progress', bar='Pending Intents', percent=round(count * 100 / common.java_files.__len__()))
+        current_file = j
+        try:
+            tree = parser.parse_file(j)
+        except ValueError as e:
+            continue
+        # TODO - Need to add scanning of the imports, to see if Intent or PendingIntent is extended, was working on it,
+        # but the one issue where it arose was non-trivial, so I gave up for now
+
+        try:
+            for import_decl in tree.import_declarations:
+                if 'PendingIntent' in import_decl.name.value:
+                    for type_decl in tree.type_declarations:
+                        if type(type_decl) is m.ClassDeclaration:
+                            for t in type_decl.body:
+                                try:
+                                    empty_intent(t, j, results)
+                                except Exception as e:
+                                    common.logger.debug("Problem in empty intent function of findPending.py: " + str(e))
+                                    common.parsingerrors.add(str(current_file))
+                                for f in t._fields:
+                                    # dynamically parse each token where is f is the field and t is the token
+                                    try:
+                                        recurse(f, t, results)
+                                    except Exception as e:
+                                        common.logger.debug("Problem in recurse function of findPending.py: " + str(e))
+                                        common.parsingerrors.add(str(current_file))
+
+        except Exception:
+            common.logger.debug("No type declarations: " + str(j))
+            report.write("parsing error-issues-list", str(current_file), "strong")
+    queue.put(results)
+    return
+
+
+def empty_intent(t, filename, results):
+    if type(t) is m.MethodDeclaration:
+        for x in method_list:
+            if str(x) in str(t):
```

Comment:
it's kind of hard to diffrentiate what x and t mean here
would recommend renaming x to method_obj (or something similar)  and t to method_declaration. Would also rename the function to something like `check_method_declaration_for_intents`, and mention in a comment somewhere that it modifies the result list (I think it does anyway?)